managed-ledger/src/main/java/org/apache/bookkeeper/mledger/ManagedLedger.java

```diff
@@ -193,6 +193,7 @@
      * @return the new NonDurableCursor
      */
     ManagedCursor newNonDurableCursor(Position startCursorPosition) throws ManagedLedgerException;
+    ManagedCursor newNonDurableCursor(Position startPosition, String subscriptionName) throws ManagedLedgerException;
```

Comment:
I think that the correct action here would be to modify existing `newNonDurableCursor` instead of adding a new one, but I did that way in order to leave this change as easy as possible, please let me know what you think.